go/vt/wrangler/vdiff.go

```diff
@@ -466,6 +478,64 @@ func findPKs(table *tabletmanagerdatapb.TableDefinition, targetSelect *sqlparser
 	return orderby, nil
 }
 
+// getColumnCollations determines the proper collation to use for each
+// column in the table definition leveraging MySQL's collation inheritence
+// rules.
+func getColumnCollations(table *tabletmanagerdatapb.TableDefinition) (map[string]collations.Collation, error) {
+	collationEnv := collations.Local()
+	createstmt, err := sqlparser.Parse(table.Schema)
+	if err != nil {
+		return nil, err
+	}
+	createtable, ok := createstmt.(*sqlparser.CreateTable)
+	if !ok {
+		return nil, vterrors.Wrapf(err, "invalid table schema %s for table %s", table.Schema, table.Name)
+	}
+	tableschema, err := schemadiff.NewCreateTableEntity(createtable)
```

Comment:
I'm not sure this is an idiomatic use of `schemadiff`, but perhaps this is an emergent pattern.

Reply:
Yeah, it's definitely outside of the main use case. But it seemed to be the best option available for this particular need. If you can think of a better alternative then I'm happy to use that. I even considered adding the GetColumnCollations() function to schemadiff but that definitely felt wrong to me so I left the usage in vdiff1. It's also worth noting that vdiff1 is already deprecated in v16 so it won't be around forever (even considered NOT fixing it here).